src/psyclone/domain/lfric/kernel/lfric_kernel_metadata.py

```diff
@@ -0,0 +1,506 @@
+            if form == "gh_scalar":
+                arg = ScalarArg.create_from_psyir(meta_arg)
+            elif form == "gh_operator":
+                arg = OperatorArg.create_from_psyir(meta_arg)
+            elif form == "gh_columnwise_operator":
+                arg = ColumnwiseOperatorArg.create_from_psyir(meta_arg)
+            elif "gh_field" in form:
+                vector_arg = "gh_field" in form and "*" in form
+                nargs = len(meta_arg.children[1].children)
+                intergrid_arg = False
+                if nargs == 5:
+                    fifth_arg = meta_arg.children[1].children[4]
+                    if fifth_arg.children[0].string == "mesh_arg":
+                        intergrid_arg = True
+
+                if intergrid_arg and vector_arg:
+                    arg = InterGridVectorArg.create_from_psyir(meta_arg)
+                elif intergrid_arg and not vector_arg:
+                    arg = InterGridArg.create_from_psyir(meta_arg)
+                elif vector_arg and not intergrid_arg:
+                    arg = FieldVectorArg.create_from_psyir(meta_arg)
+                else:
+                    arg = FieldArg.create_from_psyir(meta_arg)
+            else:
+                raise ParseError(
+                    f"Expected a 'meta_arg' entry to be a "
+                    f"field, a scalar or an operator, but found "
+                    f"'{meta_arg}'.")
+            kernel_metadata.meta_args.append(arg)
+
+        try:
+            meta_funcs = LFRicKernelMetadata._get_property(
+                spec_part, "meta_funcs")
+            args = walk(meta_funcs, Fortran2003.Ac_Value_List)
+        except ParseError:
+            meta_funcs = []
+
+        LFRicKernelMetadata.meta_reference_element = []
+        try:
+            LFRicKernelMetadata.meta_reference_element = \
+                LFRicKernelMetadata._get_property(
+                    spec_part, "meta_reference_element")
+        except ParseError:
+            pass
+        args = walk(LFRicKernelMetadata.meta_reference_element,
+                    Fortran2003.Ac_Value_List)
+        if not args:
+            LFRicKernelMetadata.meta_reference_element = []
+
+        # meta_mesh contains arguments which have
+        # properties.
+        try:
+            LFRicKernelMetadata.meta_mesh = LFRicKernelMetadata._get_property(
+                spec_part, "meta_mesh")
+        except ParseError:
+            # meta_mesh is not specified in the metadata
+            LFRicKernelMetadata.meta_mesh = []
+        return kernel_metadata
+
+    def lower_to_psyir(self):
+        '''Lower the metadata to language-level PSyIR.
+
+        :returns: metadata as stored in language-level PSyIR.
+        :rtype: :py:class:`psyclone.psyir.symbols.DataTypeSymbol`
+
+        '''
+        return DataTypeSymbol(
+            str(self.name), UnknownFortranType(self.fortran_string()))
+
+    @staticmethod
+    def _get_property(spec_part, property_name):
```

Comment:
Is this/could this method be shared with the GOcean class?

Reply:
As mentioned earlier I purposely ignored any overlap between GOcean and LFRic while getting LFRic to work to avoid fighting on multiple fronts.

Reply:
I propose to leave gocean and lfric commonality to another issue. I've created issue #1886.